sbc_compassion/models/correspondence_template.py

```diff
@@ -168,11 +172,72 @@ def _set_image(self):
+                        template.with_context(bin_size=False).template_image))
+                    template_file.flush()
+                    # Find the pattern inside the template image
+                    img = cv2.imread(template_file.name)
+                    template.page_height, template.page_width = img.shape[:2]
+                    template.qrcode_x_min = template.page_width * float(
+                        template.env['ir.config_parameter'].get_param(
+                            'qrcode_x_min'))
+                    template.qrcode_x_max = template.page_width * float(
+                        template.env['ir.config_parameter'].get_param(
+                            'qrcode_x_max'))
+                    template.qrcode_y_min = template.page_height * \
+                        float(template.env['ir.config_parameter'].get_param(
+                            'qrcode_y_min'))
+                    template.qrcode_y_max = template.page_height * \
+                        float(template.env['ir.config_parameter'].get_param(
+                            'qrcode_y_max'))
+
+    @api.depends('template_attachment_id',
+                 'pattern_image', 'template_image')
+    def _compute_template_keypoints(self):
+        """ This method computes all keypoints that can be automatically
+        detected (bluesquare and pattern_center)
+        """
+        for template in self:
+            if (template.template_image and template.pattern_image and
+                    _verify_template(template)):
+                with tempfile.NamedTemporaryFile() as template_file:
+                    template_file.write(base64.b64decode(
+                        template.with_context(bin_size=False).template_image))
+                    template_file.flush()
+                    # Find the pattern inside the template image
+                    img = cv2.imread(template_file.name)
+                    template.page_height, template.page_width = img.shape[:2]
+                    # pattern detection
+                    pattern_keypoints = pr.patternRecognition(
+                        img, template.with_context(
+                            bin_size=False).pattern_image,
+                        template.get_pattern_area())
+                    if pattern_keypoints is None:
+                        raise Warning(
+                            _("Pattern not found"),
+                            _("The pattern could not be detected in given "
+                              "template image."))
+                    # find center of the pattern
+                    pattern_center = pr.keyPointCenter(pattern_keypoints)
+                    template.pattern_center_x = pattern_center[0]
+                    template.pattern_center_y = pattern_center[1]
+                    # blue corner detection
+                    bluecorner = bcf.BlueCornerFinder(
+                        img).getIndices()
+                    template.bluesquare_x = bluecorner[0]
```

Comment:
Since `bluesquare_x` and `bluesquare_y` are computed in this method, they must also be attached to the method by adding `compute='_coompute_template_keypoints', store=True` to the field definition.